app/knight.py

```diff
@@ -0,0 +1,40 @@
+class Knight:
+    protection = 0
+
+    def __init__(
+        self,
+        name: str,
+        power: int,
+        hp: int,
+        armour: list,
+        weapon: dict,
+        potion: dict
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+
+    def set_armour(self) -> None:
+        if self.armour:
+            for equipment in self.armour:
+                self.protection += equipment.get("protection")
+
+    def set_weapon(self) -> None:
+        self.power += self.weapon.get("power")
+
+    def set_potion(self) -> None:
+        if self.potion:
+            if "power" in self.potion.get("effect"):
+                self.power += self.potion.get("effect").get("power")
+            if "protection" in self.potion.get("effect"):
+                self.protection += self.potion.get("effect").get("protection")
+            if "hp" in self.potion.get("effect"):
+                self.hp += self.potion.get("effect").get("hp")

```

Comment:
Regarding these three if conditions, avoid using them in the following way:
- Create a list of `effects` where you will have names of all effects;
- Iterate through this list;
- Use the name of the effect together with `getattr` and `setattr` to update the values.

It will allow to make code more Pythonic and avoid using code duplication.